From 8561958ccfd9b3a742a8742cc5931e9a6f6a7814 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jos=C3=A9=20Exp=C3=B3sito?= <jexposit@redhat.com>
Date: Thu, 25 Apr 2024 11:47:31 +0200
Subject: [PATCH] meson: Disable proc_macro2 unestable features
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

proc-macro2 >= 1.0.80 uses proc_macro_byte_character [1] and
proc_macro_c_str_literals [2].
Disable them until they are available.

[1] https://github.com/rust-lang/rust/issues/115268
[2] https://github.com/rust-lang/rust/issues/119750
Signed-off-by: José Expósito <jexposit@redhat.com>
---
 subprojects/packagefiles/proc-macro2/meson.build | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/subprojects/packagefiles/proc-macro2/meson.build b/subprojects/packagefiles/proc-macro2/meson.build
index b39d76e80c825..5bd635c80a389 100644
--- a/subprojects/packagefiles/proc-macro2/meson.build
+++ b/subprojects/packagefiles/proc-macro2/meson.build
@@ -41,6 +41,12 @@ endif
 if rc.version().version_compare('< 1.57')
   rust_args += ['--cfg', 'no_is_available']
 endif
+if rc.version().version_compare('< 1.79')
+  rust_args += [
+    '--cfg', 'no_literal_byte_character',
+    '--cfg', 'no_literal_c_string',
+    ]
+endif
 
 u_ind = subproject('unicode-ident').get_variable('lib')
 
-- 
GitLab

